### PR TITLE
DFC-322 add analytics package (GA4 implementation)

### PR DESCRIPTION
**Pull request by di-fabs**
## What?

- Add new analytics package into package.json
- update copy-assets build command
- create new GTM (google tag manager) constants
- create a new function (setGTM) which create new locales variables
- add call to analytics script init function
- add a new nunjuck components: ga4-opl. It's adding a new javascript code which call the new On page load tracker init function.
- add ga4-opl call to all the views.

## Why?
Integrate the new analytics package (https://www.npmjs.com/package/@govuk-one-login/frontend-analytics) into the frontend repo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,16 @@ brew install aws-sam-cli --build-from-source
...
 
+### Set the analytics environment variables
+
+Add environment variables to your .env file:
+
+UNIVERSAL_ANALYTICS_GTM_CONTAINER_ID="GTM-XXXXXXX"
+GOOGLE_ANALYTICS_4_GTM_CONTAINER_ID="GTM-XXXXXXX"
+GA4_DISABLED="false"
+UA_DISABLED="true"
+ANALYTICS_COOKIE_DOMAIN="localhost"
```
Comment (ethanmills) [inline README.md:32]: Could this be added to the `.env.sample` file please?

File: src/components/errors/404.njk [hunk 2]
```diff
@@ -15,4 +15,6 @@
         "type": "Submit",
         "href": "error.error404.content.govUKHomepageButtonHref" | translate
     }) }}
+    
+    {{ga4OnPageLoad({ nonce: scriptNonce,statusCode:"404",englishPageTitle:"error.error404.title" | translate, taxonomyLevel1: "account", taxonomyLevel2:"orchestration", contentId:"001",loggedInStatus:true,dynamic:false})}}
```
Comment (ethanmills) [inline src/components/errors/404.njk:19]: We don't know if the user is logged in here or not. Is `true` ok for this?

**Review by ethanmills — COMMENTED** [review]
Would you mind squashing the prettier commits please?

File: src/components/errors/404.njk (hunk 2 above)
Comment (di-fabs) [reply src/components/errors/404.njk:19]: Can you confirm me this local variable `res.locals.persistentSessionId` can be a good way to know if the visitor is logged in or no?

Comment (ethanmills) [reply src/components/errors/404.njk:19]: No, this is set on their first interaction with OL and does not change, regardless of login state. I think on this page (and the 500 page) it's fundamentally unknowable if the user is logged in or not

Comment (di-fabs) [reply src/components/errors/404.njk:19]: Following this information, I have to set the value to "undefined". It's the value Data & Analytics team wants when "we don't know".
Thanks @ethanmills

Comment (di-fabs) [reply src/components/errors/404.njk]: At the end, we decide to put false and create the jira ticket https://govukverify.atlassian.net/browse/DFC-414 to improve our implementation

File: README.md (hunk 1 above)
Comment (di-fabs) [reply README.md:32]: I updated .env.sample

**Review by ethanmills — APPROVED** [review]

**Review by ethanmills — APPROVED** [review]

**Pull request merged**